neuro_evolution_ctrnn/brain_visualizer/position.py

```diff
@@ -0,0 +1,123 @@
+        fpath = "position.json"
+        if os.path.isfile(fpath) and os.path.getsize(fpath) == 0:
+            numpyData = nx.spring_layout(G, k=5, weight="myweight", iterations=100)
+            with open("position.json", "w") as write_file:
+                json.dump(numpyData, write_file, cls=NumpyArrayEncoder)
+                write_file.close()
+
+        with open("position.json", "r") as read_file:
+            decodedArray = json.load(read_file)
+            read_file.close()
+        # print(decodedArray)
+        initialPosDict = {}
+        for i in range(30):
+            value = decodedArray[str(i)]
+            initialPosDict[i] = value
+
+        # pos ist ein Dictionary mit Werten in einem Koordintensystem mit Ursprung in der Mitte und Achsenlängen von -1 bis 1
+        #pos = nx.spring_layout(G, pos=initialPosDict, weight="myweight")
+        # pos2 ist ein Dictionary mit Werten in einem Koordintensystem mit Ursprung in der Mitte und Achsenlängen von 0 bis h/2-100; h = höhe des Bildschirms
+        pos2 = nx.spring_layout(G, k=2, pos=initialPosDict, weight="myweight", scale=h / 2 - 50)
+
+
+        graphPositionsDict = {}
+        for each in pos2:
+            position0 = pos2[each]
+            pos_x0 = int(position0[0] + (w / 2))
+            if pos_x0 > (w/2):
+                pos_x0 = pos_x0 + 50
+            if pos_x0 < (w/2):
+                pos_x0 = pos_x0 - 50
+            pos_y0 = int(position0[1] + (h / 2)) + 60
+            graphPositionsDict[each] = [pos_x0, pos_y0]
+
+        return graphPositionsDict
+
+
+    def clearJSON(self):
+        open("position.json", "w").close()
+        print("exiting")
+
+
+    def getInputOutputPositions(self, numberNeurons, inputOrOutput):
+        PositionsDict = {}
+        if inputOrOutput == "input":
+            x = ((1 * self.w) / 12)
+            x2 = ((1 * self.w) / 18)
+            x3 = ((2 * self.w) / 18)
+        elif inputOrOutput == "output":
+            x = ((11 * self.w) / 12)
+            x2 = ((16 * self.w) / 18)
+            x3 = ((17 * self.w) / 18)
+
+        for i in range(numberNeurons):
+            if ((self.h - 100) / (numberNeurons * (self.neuronRadius*2))) > 1:
+                x_pos = x
+                y_pos = (50 + (self.h / 2)) - ((numberNeurons * (self.neuronRadius))) + (i * (self.neuronRadius*2))
+                PositionsDict[i] = [x_pos, y_pos]
+            else:
+                if i % 2:  # ungerade
+                    x_pos = x2
+                    y_pos = (50 + (self.h / 2)) - ((numberNeurons * self.neuronRadius) / 2) + (i * self.neuronRadius)
+                    PositionsDict[i] = [x_pos, y_pos]
+                else:  # gerade
+                    x_pos = x3
+                    y_pos = (50 + (self.h / 2)) - ((numberNeurons * self.neuronRadius) / 2) + (i * self.neuronRadius)
+                    PositionsDict[i] = [x_pos, y_pos]
+        return PositionsDict
+
```

Comment:
Ich denke folgender algorithmus wäre sinnvoll: 

1. Man gibt den input neuronen positionen
1. Man gibt den output neuronen positionen
1. Man platziert die inneren neuron in der mitte des bildschirmes (zufällig, oder auf gerader linier, oder im Raster. Wie genau ist eigentlich, glaube ich)
1. Man machte eine große liste, in der ALLE neuronen sind (also input, output und innere) 
1. Man machte eine große liste, in der nur Input und output neuronen sind
1. man ruft spring_layout auf mit allen neuronen, aber übergibt die input und output-neuronen als "fixed"-parameter. 
1. man platziert alle neuronen mit den werten, die spring-layout liefert (input und output sollten hier den gleichen wert haben wir am anfang auch)


https://networkx.github.io/documentation/networkx-1.9/reference/generated/networkx.drawing.layout.spring_layout.html

Aber ich glaube es wäre ziemlich aufwendig das jetzt noch umzubauen, und ich glaube dass der vorteil nicht so super groß wäre. Daher denke ich, man kann sich das sparen